src/ShopCertification.php

```diff
@@ -117,11 +118,20 @@ public function setOrderId($orderId)
     /**
      * @param string $productItemId Must be same as ITEM_ID provided in the Heureka XML feed.
      *
-     * @return self
+     * @return ShopCertification
+     * @throws DuplicateProductItemIdException
      */
     public function addProductItemId($productItemId)
     {
-        $this->productItemIds[] = (string)$productItemId;
+        $productItemId = (string)$productItemId;
+
+        if (isset($this->productItemIds[$productItemId])) {
```

Comment:
takhle to podle me nemuzem udelat (sem si predtim neuvedomil, upozornil me na to kahis), protoze co kdyz si nekdo koupi vic kusu? tak to tam do toho API nasazi ty lidi nejakym foreachem a to pak zarve. Spis bych ty ITEMIDs jen vkladal do klice toho pole, bez te kontroly jestli uz to tam existuje, cimz se pomerne efektivne odstrani duplicity jeste predtim, nez to poslou k nam na API. Stejne neresime kolik kusu ten uzivatel zakoupil, takze to bude asi nejlepsi reseni. Takhle by si oni museli poresit duplicity na sve strane coz implementatorum pridela akorat zbytecne praci